torch/csrc/api/include/torch/data/samplers/sequential.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <torch/data/samplers/base.h>
+
+#include <ATen/optional.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <random>
+#include <vector>
+
+namespace torch {
+namespace data {
+namespace samplers {
+
+class SequentialSampler : public Sampler<> {
+ public:
+  SequentialSampler(size_t dataset_size) : dataset_size_(dataset_size) {}
+
+  void reset() override {
+    index_ = 0;
+  }
```

Comment:
That would allocate a whole new huge vector in the case of `RandomSampler`. Do you think that's a good idea?

Reply:
That's what we do in Python and it's fine. It happens once per epoch, and you can amortize it to a constant cost per iteration if you really want.

Reply:
Now that you reallocate `RandomSampler` permutation it's actually easy to do what I proposed 😄 We should really drop the sampler immediately once the data loader is exhausted since the permutation might take some space in memory.